default.htm

```diff
@@ -1,255 +1,527 @@
+                    yellowWires = Number(document.querySelector('input[name="button_yellowwires"]:checked').value),
+                    blackWires = Number(document.querySelector('input[name="button_blackwires"]:checked').value),
+                    whiteWires = Number(document.querySelector('input[name="button_whitewires"]:checked').value),
+                    serial = Boolean(document.querySelector('input[name="button_serial"]:checked').value),
+                    lastwire = document.querySelector('input[name="button_lastwire"]:checked').value;
+
+ 				let totalcount = redWires + blueWires + yellowWires + blackWires + whiteWires;
+
+				var white = "white", yellow = "yellow", blue = "blue", red = "red", black = "black",
+					last = "last", first = "first", second = "2nd", fourth = "4th", third = "3rd";
+	
+				let result = "";
+
+				switch (totalcount) {
+					case 3:
+						{
+							if (redWires == 0) {
+								result = second;
+								break;
+							}
+                            if (lastwire == white) {
+								result = last;
+								break;
+							}
+							if (blueWires > 1) {
+								result = last + " " + blue;
+								break;
+							}
+							result = last;
+							break;
+						}
+					case 4:
+						{
+							if (redWires > 1 && serial) {
+								result = last + " " + red;
+								break;
+							}
+                            if (lastwire == yellow && redWire == 0) {
+								result = first;
+								break;
+							}
+							if (blueWires == 1) {
+								result = first;
+								break;
+							}
+							if (yellowWires > 1) {
+								result = last;
+								break;
+							}
+							result = second;
+							break;
+						}
+					case 5:
+						{
+                            if (lastwire == black && serial) {
+								result = fourth;
+								break;
+							}
+							if (redWires == 1 && yellowWires > 1) {
+								result = first;
+								break;
+							}
+							if (blackWires == 0) {
+								result = second;
+								break;
+							}
+							result = first;
+							break;
+						}
+					case 6:
+						{
+							if (yellowWires == 0 && serial) {
+								result = third;
+								break;
+							}
+							if (yellowWires == 1 && whiteWires > 1) {
+								result = fourth;
+								break;
+							}
+							if (redWires == 0) {
+								result = last;
+								break;
+							}
+
+							result = fourth;
+							break;
+						}
+					default:
+						result = "Invalid input";
+						break;
 				}
 
-				function jump(e) {
-					if ("1234567890".includes(e.key)) {
-						e.preventDefault();
-						e.stopPropagation();
-					}
-					if ("12345".includes(e.key)) {
-						document.getElementById("letters" + e.key).focus();
-					}
+                document.getElementById("rWires").innerHTML = "Total wires: " + totalcount + ". Cut the " + result + " wire."; 
+			}
+
+			function jump(e) { //This is specific to the password solver? Should be fine for now but if there are additional text inputs this may need context.
```

Comment:
Yup I haven't got around to making their names more elaborate now that there are other kinds of solvers.